docs/notebooks/verification_gridded_cosmo.ipynb

```diff
@@ -0,0 +1,3097 @@
+    "}\n",
+    "\n",
+    "# Colorblind-friendly colormap for 2D plots\n",
+    "COLORMAP = \"viridis\"\n",
+    "\n",
+    "\n",
+    "def save_plot(fig, name, time=None):\n",
+    "    \"\"\"Helper function to save plots consistently\"\"\"\n",
+    "    if time is not None:\n",
+    "        name = f\"{name}_{time.dt.strftime('%Y%m%d_%H').values}\"\n",
+    "    fig.savefig(f\"plots/{name}.pdf\", bbox_inches=\"tight\", dpi=300)\n",
+    "    # fig.savefig(f\"plots/{name}.png\", bbox_inches=\"tight\", dpi=300)\n",
+    "\n",
+    "\n",
+    "def export_table(df, name, caption=\"\"):\n",
+    "    \"\"\"Helper function to export tables consistently\"\"\"\n",
+    "    # Export to LaTeX with caption\n",
+    "    latex_str = df.to_latex(\n",
+    "        float_format=\"%.4f\", caption=caption, label=f\"tab:{name}\"\n",
+    "    )\n",
+    "    with open(f\"tables/{name}.tex\", \"w\") as f:\n",
+    "        f.write(latex_str)\n",
+    "\n",
+    "    # Export to CSV\n",
+    "    df.to_csv(f\"tables/{name}.csv\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d726501f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# REMOVE THIS WORKAROUND ONCE THE ZARR IS FIXED\n",
+    "ds = xr.open_zarr(PATH_GROUND_TRUTH)\n",
+    "ds = ds.drop_vars([\n",
+    "    \"state_feature_units\",\n",
+    "    \"state_feature_long_name\",\n",
+    "    \"state_feature_source_dataset\",\n",
+    "])\n",
+    "state_mean = ds.state__train__mean\n",
+    "state_std = ds.state__train__std"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "86ebba4b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds_ml = xr.open_zarr(PATH_ML)\n",
+    "ds_ml[\"state\"] = ds_ml[\"state\"] * state_std + state_mean\n",
+    "ds_ml = ds_ml.sel(state_feature=list(VARIABLES_ML.keys()))\n",
+    "ds_ml = ds_ml.sel(y=slice(*Y), x=slice(*X))\n",
+    "ds_ml = ds_ml.sel(start_time=slice(*START_TIMES))\n",
+    "for feature in ds_ml.state_feature.values:\n",
+    "    ds_ml[VARIABLES_ML[feature]] = ds_ml[\"state\"].sel(state_feature=feature)\n",
+    "forecast_times = (\n",
+    "    ds_ml.start_time.values[:, None] + ds_ml.elapsed_forecast_duration.values\n",
+    ")\n",
+    "ds_ml = ds_ml.assign_coords(\n",
+    "    forecast_time=((\"start_time\", \"elapsed_forecast_duration\"), forecast_times)\n",
+    ")\n",
+    "ds_ml = ds_ml.drop_vars([\"state\", \"state_feature\", \"time\"])\n",
+    "ds_ml = ds_ml.transpose(\"start_time\", \"elapsed_forecast_duration\", \"x\", \"y\")\n",
+    "ds_ml = ds_ml[\n",
+    "    [\n",
+    "        \"start_time\",\n",
+    "        \"elapsed_forecast_duration\",\n",
+    "        \"x\",\n",
+    "        \"y\",\n",
+    "        *VARIABLES_ML.values(),\n",
+    "    ]\n",
+    "]\n",
+    "ds_ml = ds_ml.isel(elapsed_forecast_duration=ELAPSED_FORECAST_DURATION)\n",
+    "ds_ml = ds_ml.compute()\n",
+    "\n",
+    "ds_ml_first_timestep = (\n",
```

Comment:
Now this variable, that is for time step 1, is used for quite a lot of plots and calculations below (if I don't misunderstand). Is the idea that I should change `elapsed_forecast_duration=0` here to match the lead time that I care about, and want to make plots for?

Reply:
i changed this so that the whole notebook evaluates now all `elapsed_forecast_duration`s the user selected. this means the user is now expected to evaluate their choices and make sure the memory/runtime requirements are feasible for their system. but scientifically this makes now much more sense, because no arbitrary subsetting to time_step 1 is required.